### ISSUE TITLE
Warning message when using custom loss function

### ISSUE THREAD

**Issue by notprime** [tensorflow/tensorflow#48825]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): YES
- OS Platform and Distribution: UBUNTU 18.04
- TensorFlow installed from: CONDA
- TensorFlow version: 2.4.1
- Python version: 3.8
- CUDA/cuDNN version: 11.2
- GPU model and memory: 4 x TITAN Xp 12GB

----------------------

Greetings from Italy!
I coded a simple autoencoder (I used one of the stock example script you can find on TF) just to see if I was able to implement a custom loss function. This is the code (jupyter notebook file): 

https://github.com/notprime/autoencoder/blob/main/autoencoder.ipynb

Basically, that **`function`** in `[3]` is something I didn't code, and its vectorialization is pretty hard. Therefore, when I get the hidden output tensor, I convert it into a numpy array, compute the matrix I need, convert it back into a TF tensor, and I compute the custom loss function.

When I train the model, I get the following error:
```
WARNING:tensorflow:AutoGraph could not transform <function compute_loss at 0x7f0e688fbe50> and will run it as-is.
Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output.
Cause: module 'gast' has no attribute 'Index'
```

Is this something I should worry about?
Moreover, how does TF handle the automatic-differentiation of this `compute_loss` function (which calls in every for loop the other ** `function`**? This is something I'd really like to better understand.
If you also have some advices on how optimize the code let me know, this is really my first time using TF. Maybe I can convert those function into TF function using the methods `tf.autograph` or `tf.autograph.to_graph`?

Thanks in advance!

**Comment by saikumarchalla** [comment]
@notprime  Could you please confirm what version of gast are you using? Please try with gast 0.3.3 for TF version 2.4. Thanks!
